src/GeneratorTwigHelper.php

```diff
@@ -31,15 +31,29 @@ public function getEntityFieldPrintCode($entity, $field): string
         $printCode = $entity.'.'.str_replace('_', '', $twigField);
 
         switch ($field['type']) {
+            case 'datetimetz_immutable':
+            case 'datetimetz':
+                $printCode .= ' ? '.$printCode.'|date(\'Y-m-d H:i:s T\') : \'\'';
+                break;
+            case 'datetime_immutable':
             case 'datetime':
                 $printCode .= ' ? '.$printCode.'|date(\'Y-m-d H:i:s\') : \'\'';
                 break;
+            case 'dateinterval':
+                $printCode .= ' ? '.$printCode.'.format(\'%y years, %m month, %d days\') : \'\'';
+                break;
+            case 'date_immutable':
             case 'date':
                 $printCode .= ' ? '.$printCode.'|date(\'Y-m-d\') : \'\'';
                 break;
+            case 'time_immutable':
             case 'time':
                 $printCode .= ' ? '.$printCode.'|date(\'H:i:s\') : \'\'';
                 break;
+            case 'json':
+            case 'json_array':
+                $printCode .= ' is iterable ? '.$printCode.'|join(\', \') : '.$printCode;
+                break;
```

Comment:
Interesting! From looking at the docs, it seems like the value in this field will always be an `array` or `null`. Which case were you covering with the `is iterable`?

And, because the array can be an associative array, the `|join` won't look quite right in those cases. We could use `dump()` WDYT?

Reply:
I'll check it again. I thought there could be string value.

`dump()` is a good idea, but will it work without debug package?

Reply:
It believe that it DOES work without `debug` - it reverts to `var_dump()`. We can add a `json` field to our entity in the functional test to be sure.

Reply:
Oh, you know what? Let's use `|json_encode`. That will print nicely, and make it very obvious that this is a JSON field. And, it will definitely work :).

@sadikoff if you agree, can you make that change? Also, if you rebase from the latest master, the tests should be closer to passing.